R/test-compiled-code.R

```diff
@@ -21,10 +26,66 @@ expect_cpp_tests_pass <- function(package) {
     }
   )
 
-  # Drop first line of output (it's jut a '####' delimiter)
-  info <- paste(output[-1], collapse = "\n")
+  report <- xml2::read_xml(paste(output, collapse = "\n"))
```

Comment:
Should this be wrapped in `tryCatch()` in case Catch produces invalid XML (e.g. maybe something goes wrong during a segfault)?

Reply:
sure, that is probably safest

Reply:
Actually, I guess if the XML is invalid then xml2 should fail, which is what we want in this case I think?

Reply:
So this happened to me yesterday, and I am not sure of the best way to handle it. Even if we throw a normal error here it ends up being caught in the testthat machinery, but not shown, so you see there was a warning and an error, but not what they were.

I guess we could throw a registered error like we do below...